src/common/redis_module/ray_redis_module.cc

```diff
@@ -801,54 +820,78 @@ int ResultTableLookup_RedisCommand(RedisModuleCtx *ctx,
 int TaskTableWrite(RedisModuleCtx *ctx,
                    RedisModuleString *task_id,
                    RedisModuleString *state,
-                   RedisModuleString *node_id,
+                   RedisModuleString *local_scheduler_id,
                    RedisModuleString *task_spec) {
-  /* Pad the state integer to a fixed-width integer, and make sure it has width
-   * less than or equal to 2. */
-  state = NormalizeTaskState(ctx, state);
-  if (state == NULL) {
+  /* Format the state as a fixed width string of length 2 so that we can treat
+   * it as a bit mask and pattern match it with PSUBSCRIBE. This is done in
+   * redis_task_table_subscribe in redis.cc. */
+  RedisModuleString *formatted_state = NormalizeTaskState(ctx, state);
```

Comment:
Now that we're using flatbuffers, I think we can simplify/get rid of the `NormalizeTaskState` function (just check if it can be cast to an integer).